kratos/processes/apply_periodic_boundary_condition_process.cpp

```diff
@@ -0,0 +1,362 @@
+        rMatrix(3,0) = 0.0;
+        rMatrix(3,1) = 0.0;
+        rMatrix(3,2) = 0.0;
+        rMatrix(3,3) = 1.0;
+    }
+
+    void ApplyPeriodicConditionProcess::CalculateRotationMatrix(const double Theta, MatrixType& rMatrix )
+    {
+        std::vector<double> U(3); // normalized axis of rotation
+        // normalizing the axis of rotation
+        double norm = 0.0;
+        for (IndexType d = 0; d < 3; ++d)
+            norm += mAxisOfRotationVector[d] * mAxisOfRotationVector[d];
+        norm = sqrt(norm);
+        for (IndexType d = 0; d < 3; ++d)
+            U[d] = mAxisOfRotationVector[d] / norm;
+
+        // Constructing the transformation matrix
+        const double x1 = mCenterOfRotation[0];
+        const double y1 = mCenterOfRotation[1];
+        const double z1 = mCenterOfRotation[2];
+
+        const double a = U[0];
+        const double b = U[1];
+        const double c = U[2];
+
+        const double t2 = std::cos(Theta);
+        const double t3 = std::sin(Theta);
+        const double t4 = a * a;
+        const double t5 = b * b;
+        const double t6 = c * c;
+        const double t7 = a * b;
+        const double t8 = t5 + t6;
+        const double t9 = std::abs(t8) < std::numeric_limits<double>::epsilon() ? 1.0e8 : 1.0 / t8;
+        const double t10 = a * c;
+        const double t11 = b * t3;
+        const double t12 = a * t3 * t5;
+        const double t13 = a * t3 * t6;
+        const double t14 = b * c * t2;
+        rMatrix(0,0) = t4 + t2 * t8;
+        rMatrix(0,1) = t7 - c * t3 - a * b * t2;
+        rMatrix(0,2) = t10 + t11 - a * c * t2;
+        rMatrix(0,3) = x1 - t4 * x1 - a * b * y1 - a * c * z1 - b * t3 * z1 + c * t3 * y1 - t2 * t5 * x1 - t2 * t6 * x1 + a * b * t2 * y1 + a * c * t2 * z1;
+        rMatrix(1,0) = t7 + c * t3 - a * b * t2;
+        rMatrix(1,1) = t9 * (t2 * t6 + t5 * t8 + t2 * t4 * t5);
+        rMatrix(1,2) = -t9 * (t12 + t13 + t14 - b * c * t8 - b * c * t2 * t4);
+        rMatrix(1,3) = -t9 * (-t8 * y1 + t2 * t6 * y1 + t5 * t8 * y1 + a * b * t8 * x1 - b * c * t2 * z1 + b * c * t8 * z1 - a * t3 * t5 * z1 - a * t3 * t6 * z1 + c * t3 * t8 * x1 + t2 * t4 * t5 * y1 - a * b * t2 * t8 * x1 + b * c * t2 * t4 * z1);
+        rMatrix(2,0) = t10 - t11 - a * c * t2;
+        rMatrix(2,1) = t9 * (t12 + t13 - t14 + b * c * t8 + b * c * t2 * t4);
+        rMatrix(2,2) = t9 * (t2 * t5 + t6 * t8 + t2 * t4 * t6);
+        rMatrix(2,3) = -t9 * (-t8 * z1 + t2 * t5 * z1 + t6 * t8 * z1 + a * c * t8 * x1 - b * c * t2 * y1 + b * c * t8 * y1 + a * t3 * t5 * y1 + a * t3 * t6 * y1 - b * t3 * t8 * x1 + t2 * t4 * t6 * z1 - a * c * t2 * t8 * x1 + b * c * t2 * t4 * y1);
+        rMatrix(3,0) = 0.0;
+        rMatrix(3,1) = 0.0;
+        rMatrix(3,2) = 0.0;
+        rMatrix(3,3) = 1.0;
+    }
+
+    void ApplyPeriodicConditionProcess::TransformNode(array_1d<double, 3 >& rCoordinates, array_1d<double, 3 >& rTransformedCoordinates)
+    {
+        std::vector<double> original_node(4, 0.0f);
```

Comment:
Unnecessary type conversion
```suggestion
        std::vector<double> original_node(4, 0.0);
```